File: client/src/main/java/com/jraska/github/client/ui/UriHandlerActivity.java
package com.jraska.github.client.ui;

import android.os.Bundle;
import android.support.annotation.Nullable;
import com.jraska.github.client.UriHandlerViewModel;

public class UriHandlerActivity extends BaseActivity {
  @Override protected void onCreate(@Nullable Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);

    UriHandlerViewModel viewModel = viewModel(UriHandlerViewModel.class);
    viewModel.handleDeepLink(getIntent().getData().toString());
    finish();
  }
}

File: client/src/main/java/com/jraska/github/client/ui/UriHandlerActivity.kt
package com.jraska.github.client.ui

import android.os.Bundle
import com.jraska.github.client.core.android.UriHandlerViewModel
import com.jraska.github.client.inputUrl
import com.jraska.github.client.core.android.viewModel

class UriHandlerActivity : BaseActivity() {
  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)

    val viewModel = viewModel(UriHandlerViewModel::class.java)
    viewModel.handleDeepLink(inputUrl())
    finish()
  }
}
